internal/controller/pipelinerollout_controller.go

```diff
@@ -844,6 +802,88 @@ func (r *PipelineRolloutReconciler) makePipelineDefinition(
 	}, nil
 }
 
+// the following functions enable PipelineRolloutReconciler to implement progressiveController interface
+func (r *PipelineRolloutReconciler) listChildren(ctx context.Context, rolloutObject RolloutObject, labelSelector string, fieldSelector string) ([]*kubernetes.GenericObject, error) {
+	pipelineRollout := rolloutObject.(*apiv1.PipelineRollout)
+	return kubernetes.ListLiveResource(
+		ctx, r.restConfig, common.NumaflowAPIGroup, common.NumaflowAPIVersion, "pipelines",
+		pipelineRollout.Namespace, labelSelector, fieldSelector)
+}
+
+func (r *PipelineRolloutReconciler) createBaseChildDefinition(rolloutObject RolloutObject, name string) (*kubernetes.GenericObject, error) {
+	pipelineRollout := rolloutObject.(*apiv1.PipelineRollout)
+	pipelineLabels, err := basePipelineLabels(pipelineRollout)
+	if err != nil {
+		return nil, err
+	}
+	return r.makePipelineDefinition(pipelineRollout, name, pipelineLabels)
+}
+
+func (r *PipelineRolloutReconciler) getCurrentChildCount(rolloutObject RolloutObject) (int32, bool) {
+	pipelineRollout := rolloutObject.(*apiv1.PipelineRollout)
+	if pipelineRollout.Status.NameCount == nil {
+		return int32(0), false
+	} else {
+		return *pipelineRollout.Status.NameCount, true
+	}
+}
+
+func (r *PipelineRolloutReconciler) updateCurrentChildCount(ctx context.Context, rolloutObject RolloutObject, nameCount int32) error {
+	pipelineRollout := rolloutObject.(*apiv1.PipelineRollout)
+	pipelineRollout.Status.NameCount = &nameCount
+	return r.updatePipelineRolloutStatus(ctx, pipelineRollout)
+}
+
+// increment the child count for the Rollout and return the count to use
+func (r *PipelineRolloutReconciler) incrementChildCount(ctx context.Context, rolloutObject RolloutObject) (int32, error) {
+	currentNameCount, found := r.getCurrentChildCount(rolloutObject)
+	if !found {
+		currentNameCount = int32(0)
+		err := r.updateCurrentChildCount(ctx, rolloutObject, int32(0))
+		if err != nil {
+			return int32(0), err
+		}
+	}
+
+	// TODO: why in PipelineRolloutReconciler.calPipelineName() do we only update the Status subresource when it's 0?
```

Comment:
was wondering why we do the Update in Kubernetes [here](https://github.com/numaproj/numaplane/blob/main/internal/controller/pipelinerollout_controller.go#L797) but not [here](https://github.com/numaproj/numaplane/blob/main/internal/controller/pipelinerollout_controller.go#L805)

Maybe we always want to increment the `NameCount` in the Status before we create a Pipeline which is based off that value? Otherwise, we can look at the scenario in which we update the Pipeline name first and then fail to update the `NameCount` in Kubernetes, could we create a new Pipeline with the wrong name next time around?